### PR TITLE
[ChunkTeacher] Fix a hang

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Fix a bug where CT would hang when the buffer was smaller than the size of the the test set chunk.

Also define a new set of tests to pretty exhaustively test CT in a wide variety of settings.

**Testing steps**
New CI

### CODE REVIEW

File: parlai/agents/test_agents/counter.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        return shared
+
+    def save(self, path=None):
+        pass
+
+    def load(self, path=None):
+        pass
+
+    def _val(self, val):
+        if isinstance(val, list):
```
Comment (emilydinan) [inline parlai/agents/test_agents/counter.py:87]: this is a bit weird so maybe add a comment here why you do this (text vs. labels)

File: parlai/core/teachers.py [hunk 2]
```diff
@@ -2514,6 +2514,11 @@ def reset(self):
             )  # reset the count of samples loaded
             self._enqueue_request()
 
+    def shutdown(self):
+        # self._drain(self.chunks)
```
Comment (emilydinan) [inline parlai/core/teachers.py:2518]: why did you comment this out?? are you sure you want to do this?

File: parlai/tasks/integration_tests/agents.py [hunk 3]
```diff
@@ -510,19 +515,14 @@ def get(self, episode_idx=0, entry_idx=0):
...
     Unique examples that load slowly.
     """
 
     def load_from_chunk(self, chunk_idx: int):
-        output = []
-        for i in range(10):
-            text = str(i + chunk_idx * 10)
-            resp = str(i + chunk_idx * 10)
-            output.append((text, resp))
         time.sleep(0.1)
```
Comment (emilydinan) [inline parlai/tasks/integration_tests/agents.py:524]: is adding this single time.sleep(0.1) enough?? why is it still considered a "slow" teacher? maybe make a diff teacher?

**Review by emilydinan — CHANGES_REQUESTED** [review]
i think you commented out `self._drain` by accident? also why did you add more `Nones` to shutdown?

File: parlai/tasks/integration_tests/agents.py (hunk 3 above)
Comment (stephenroller) [reply parlai/tasks/integration_tests/agents.py:524]: Yes, it's enough. It's been like this for a bit.

File: parlai/core/teachers.py (hunk 2 above)
Comment (stephenroller) [reply parlai/core/teachers.py:2518]: It's not necessary, need to delete

**Review by emilydinan — APPROVED** [review]

**Pull request merged**